### DISCUSSION TITLE
SchedulerWorker doesn't refresh query

### DISCUSSION THREAD

**Discussion by ognjen-it** [getredash/redash#6773] [Q&A]
Hi all,

I have a problem with the scheduler worker. It will work if I set it to run every 1 minute or 1 hour, but if I set every week it doesn't refresh query.

Configuration for scheduler:
<img width="483" alt="image" src="https://github.com/getredash/redash/assets/44586606/f19818ac-80ac-4fb1-b49a-7463c76020d5">

GUI: 
20/02/24 13:54 -> CET
<img width="1367" alt="image" src="https://github.com/getredash/redash/assets/44586606/eed44015-95d5-4476-8ddd-1ecf080a5513">

I can see it exists in redis:
<img width="1033" alt="image" src="https://github.com/getredash/redash/assets/44586606/a60bec49-f620-4367-bd15-07b2b87d8f6d">

Log:
<img width="874" alt="image" src="https://github.com/getredash/redash/assets/44586606/0383d058-eacf-442b-a1c5-9b6eb13ec821">
<img width="1421" alt="image" src="https://github.com/getredash/redash/assets/44586606/77747764-561d-4067-8ede-71ec0a1b8a02">

Is there anyone who can help me please?

I deployed Redash on Azure Kubernetes
Helm chart: https://github.com/getredash/contrib-helm-chart  Version: 3.0.1
Image version: 23.12.0-dev (I built from 23.12.0-dev tag)

**Comment by eradman** [comment]
@ognjen-it can you try the snapshot from 24.02.0? Commit 4d81c314 may solve this.

**Comment by ognjen-it** [comment]
Hi @eradman and thank you for your fast replay.
I couldn't find snapshot 24.02.0 in the repository (23.12.0 is the latest?) but I built it from the commit you provided and it's not working.

If I schedule it to run every day, I can see standard logs in scheduledworker pod:
![image](https://github.com/getredash/redash/assets/44586606/07af5228-4449-4d01-a8b5-7336defbdf70)
And in redis I can see it successfully finished:
<img width="1446" alt="image" src="https://github.com/getredash/redash/assets/44586606/8c0d8d93-14fd-4714-a661-95eb166601fb">

Job information:
<img width="1445" alt="image" src="https://github.com/getredash/redash/assets/44586606/f396b2ca-7c49-47fa-934a-e699a3beae32">

Here we can see the count of successful jobs: 3
<img width="1445" alt="image" src="https://github.com/getredash/redash/assets/44586606/b9b2803b-b679-49ab-8e6e-14e4f160e7b3">


However, If I configure a specific day (except Monday because from some reason it works on Monday), I don't see anything in logs:
![image](https://github.com/getredash/redash/assets/44586606/04941549-5467-4e05-a07c-0deea0ed4de1)

rq:finished still the same:
<img width="1652" alt="image" src="https://github.com/getredash/redash/assets/44586606/acc2c4bd-9042-4e9c-b2cf-3ca8a79e8e38">

And worker didn't change count of successful jobs:
<img width="1652" alt="image" src="https://github.com/getredash/redash/assets/44586606/2c764be0-2368-4db2-9754-27775cc5a619">

**Comment by eradman** [comment]
Even after building from source, the query hashes may be incorrect. Try downloading the attached script:
[refresh-query-hash.py.gz](https://github.com/getredash/redash/files/14421651/refresh-query-hash.py.gz)

```
gzip -d refresh-query-hash.py.gz
./refresh-query-hash-alt.py --url=http://localhost:5001 --api-key=XXXXX
```

This works by

1. Listing all queries
2. Updating each query with an empty payload to trigger the new query hash code path

**Comment by ognjen-it** [comment]
Thank you again for fast replay. Just ran script and it printed 5x strings:
![image](https://github.com/getredash/redash/assets/44586606/248adc55-088c-4751-93e2-807200256df1)

However, it didn't resolve scheduling problem.

I just tried to run redash from master (81d22f1eb2) but it also doesn't work. I still investigating where is the problem.

**Reply by ognjen-it** [reply]
@eradman just to mention that I tried to run ReDash from master branch with clean Redis and Postgres, but it's not working :(

**Comment by eradman** [comment]
Sounds like this feature just doesn't work. We'll have to investigate
